Project/Assets/ML-Agents/Examples/Crawler/Scripts/CrawlerAgent.cs

```diff
@@ -71,11 +65,28 @@ public override void Initialize()
         m_JdController.SetupBodyPart(leg3Lower);
     }
 
+    /// <summary>
+    /// Loop over body parts and reset them to initial conditions.
+    /// </summary>
+    public override void OnEpisodeBegin()
+    {
+        foreach (var bodyPart in m_JdController.bodyPartsDict.Values)
+        {
+            bodyPart.Reset(bodyPart);
+        }
+
+        //Random start rotation to help generalize
+        transform.rotation = Quaternion.Euler(0, Random.Range(0.0f, 360.0f), 0);
+
+        orientationCube.UpdateOrientation(body, target.transform);
+    }
+
     /// <summary>
     /// Add relevant information on each body part to observations.
     /// </summary>
     public void CollectObservationBodyPart(BodyPart bp, VectorSensor sensor)
     {
+        //GROUND CHECK
         sensor.AddObservation(bp.groundContact.touchingGround ? 1 : 0); // Whether the bp touching the ground
```

Comment:
I think there is a AddObservation for boolean. Maybe is this instead of casting to int ?

Reply:
changed. agreed, no reason to cast to an int.